### PR TITLE
cache control account response in heal

### PR DESCRIPTION
**Pull request by mootz12**

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "semi": true,
+  "bracketSpacing": true,
+  "singleQuote": true,
+  "trailingComma": "es5"
+}
```
Comment (mootz12) [inline .prettierrc:6]: added basic prettier config

File: package-lock.json [hunk 2]
```diff
@@ -1,24 +0,0 @@
-{
```
Comment (mootz12) [inline package-lock.json:1]: i think these were an oops. Moved this package into wrangler/package.json

File: wrangler/src/@utils/stellar-sdk-utils.js [hunk 3]
```diff
@@ -23,6 +24,24 @@ export function verifyTxSignedBy(transaction, accountID) {
...
+ * @link https://developers.stellar.org/api/resources/accounts/object/
+ */
+export async function loadAccount(publicKey) {
+  try {
+    const accountInfo = await fetch(`${HORIZON_URL}/accounts/${pubkey}`);
+    return accountInfo.json();
+  } catch {
+    throw new Error(`Unable to fetch account with publicKey ${publicKey}`);
+  }
+}
```
Comment (mootz12) [inline wrangler/src/@utils/stellar-sdk-utils.js:43]: add loadAccount util

File: wrangler/src/@utils/stellar-sdk-utils.js [hunk 4]
```diff
@@ -23,6 +24,24 @@ export function verifyTxSignedBy(transaction, accountID) {
...
+
 /**
```
Comment (mootz12) [inline wrangler/src/@utils/stellar-sdk-utils.js:45]: rest of changes was prettier

File: wrangler/src/trust/trust.js [hunk 5]
```diff
@@ -1,91 +1,35 @@
...
+  Networks,
+  BASE_FEE,
+  Operation,
+  Account,
+} from 'stellar-base';
+import { parse } from 'toml';
+import { loadAccount } from '../@utils/stellar-sdk-utils';
+import localtoml from '../turret/toml';
 
-/**
```
Comment (mootz12) [inline wrangler/src/trust/trust.js:5]: majority of these account based helpers were obsolete once the accountRecord was used

File: wrangler/src/trust/trust.js [hunk 6]
```diff
@@ -94,178 +38,167 @@ export async function getTxFunctionSigner(txfunctionhash, turretpubkey) {
...
+
+    // check the local toml to validate the new turret is part of the trust quorum and the old turret is not
+    const newtrust = checkLocalQuorum(newTurret);
+    if (!newtrust) {
+      throw `The new turret is not trusted by the local turret quorum, make sure it is added to its toml`;
+    }
+    const oldtrust = checkLocalQuorum(oldTurret);
+    if (oldtrust) {
+      throw `The old turret is still trusted by the local quorum and can't be removed`;
+    }
```
Comment (mootz12) [inline wrangler/src/trust/trust.js:109]: shifted this logic to the start

we can do it before checking any accounts, so why not

File: wrangler/src/trust/trust.js [hunk 7]
```diff
@@ -94,178 +38,167 @@ export async function getTxFunctionSigner(txfunctionhash, turretpubkey) {
...
 
-          /* generate the signature */
-          const { txFunctionSignerPublicKey, txFunctionSignerSecret } = metadata
+    // gather the old signer/turret from the control account
+    // validate it is currently a signer on the control account
+    const removeSignerKey =
+      controlAccountRecord.data[TURRET_DATA_KEY(oldTurret)];
+    const removeSignerObj = controlAccountRecord.signers.find(
+      (s) => s.key === removeSignerKey
+    );
```
Comment (mootz12) [inline wrangler/src/trust/trust.js:126]: make use of the accountRecord to capture the full removeSignerObj

File: wrangler/src/trust/trust.js [hunk 8]
```diff
@@ -94,178 +38,167 @@ export async function getTxFunctionSigner(txfunctionhash, turretpubkey) {
...
-            value: null,
-        /* add the operation to add the new managedata entry */
-        })).addOperation(Operation.manageData({
-            name: `turret.${newturretsigner}`,
-            value: newturret,
-          }))
-          .setTimeout(0)
-          /* build the transaction */
-          .build()
+export async function heal(controlAccount, oldTurret, newTurret, functionHash) {
```
Comment (mootz12) [inline wrangler/src/trust/trust.js:92]: this diff is a bit confusing with prettier and refactor.

I'd recommend reviewing the raw file as well, might make more sense.

File: .prettierrc (hunk 1 above)
Comment (silence48) [reply .prettierrc:6]: looks good

File: wrangler/src/@utils/stellar-sdk-utils.js (hunk 3 above)
Comment (silence48) [reply wrangler/src/@utils/stellar-sdk-utils.js:43]: the load account and cached account made this much more efficient.

**Pull request merged**